akka-persistence-typed/src/main/scala/akka/persistence/typed/keyvalue/javadsl/KeyValueBehavior.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2018-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.keyvalue.javadsl
+
+import java.util.Optional
+
+import akka.actor.typed
+import akka.actor.typed.BackoffSupervisorStrategy
+import akka.actor.typed.Behavior
+import akka.actor.typed.internal.BehaviorImpl.DeferredBehavior
+import akka.actor.typed.javadsl.ActorContext
+import akka.annotation.InternalApi
+import akka.persistence.typed.keyvalue.internal
+import akka.persistence.typed.keyvalue.internal._
+import akka.persistence.typed.keyvalue.scaladsl
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.SnapshotAdapter
+
+/**
+ * A `Behavior` for a persistent actor with key-value storage of its state.
+ */
+abstract class KeyValueBehavior[Command, State] private[akka] (
```

Comment:
this is the starting point for the public api for Java, everything very similar to EventSourcedBehavior

Reply:
We can see if there are anythings that should be extracted and reused, but trying too hard reducing the duplication can result in that the code is more difficult to understand. The EventSourcedBehavior machinery is already rather involved and adding another dimension to it might be "too much".

If we try to extract common pieces I think we should aim for delegation rather than extension.

Anyway, before we try any such refactoring we need tests to be sure we don't break things.